### DISCUSSION TITLE
Is it possible to reset invocation count?

### DISCUSSION THREAD

**Discussion by rawilder** [mockk/mockk#842] [Q&A] [answered]
I am trying to write some verification blocks in a shared used way, and I am having trouble verifying how many times a function gets called so I wanted to reset invocations but the only way to do that seems to be resetting the entire mock. Is there a way do this without having to re-stub the methods every time?

**Comment by Raibaz** [comment]
Not sure what you are trying to achieve, but as a general rule of thumb sharing states between tests is not a good idea, test should be completely independent on one another.

Can you please post a more complete example of the behavior you are trying to achieve?

**Reply by rawilder** [reply]
For example, I have a singleton that provides data to multiple different services. Rather than stub before every usage, I have one definition of stubs that provides general data (like a preseeded database). I use that mock a few places and want to validate the number of times a method inside it gets called but I don't want to have to re-stub or keep track of the number of invocations separately. Like mockito's clearInvocations

[javadoc.io/static/org.mockito/mockito-core/3.6.28/org/mockito/Mockito.html#clearInvocations-T...-](https://javadoc.io/static/org.mockito/mockito-core/3.6.28/org/mockito/Mockito.html#clearInvocations-T...-)

**Reply by Chrostoq** [reply, marked as answer]
You should be able to do this by calling `clearMocks` and set all parameters but `recordedCalls` to false
```
clearMocks(
    yourMockk,
    answers = false,
    recordedCalls = true,
    childMocks = false,
    verificationMarks = true,  // set this to true to have verifyAll be reset as well 
    exclusionRules = false
)
```

**Comment by k163377** [comment]
I would like this feature too.
If I want to make sure that everything works correctly on the first run and the process is skipped on the second run, I can't currently write `exactly = 0`.

Chrostoq has provided a solution, but it would be nice if this existed as a `mockk` function.
